filament/backend/src/vulkan/VulkanSwapChain.cpp

```diff
@@ -88,8 +88,8 @@ void VulkanSwapChain::present() {
                 .layerCount = 1,
         };
         mColors[mCurrentSwapIndex]->transitionLayout(cmdbuf, subresources, VulkanLayout::PRESENT);
-        mCommands->flush();
     }
+    mCommands->flush();
```

Comment:
oh there is a flush in present. so do we need the one in endFrame()?

Reply:
for reasons I don't understand yet - Yes, we do need it in `endFrame()`.  I see an increase in `cpu` usage when it's not there, on both MoltenVK and adreno. I plan to turn on a performance validation on an nvidia card to see if that tells me anything.